src/task.c

```diff
@@ -447,11 +447,21 @@ JL_NO_ASAN static void ctx_switch(jl_task_t *lastt)
 #endif
 
     int killed = jl_atomic_load_relaxed(&lastt->_state) != JL_TASK_STATE_RUNNABLE;
-    if (!t->ctx.started && !t->ctx.copy_stack) {
-        // may need to allocate the stack
-        if (t->ctx.stkbuf == NULL) {
-            t->ctx.stkbuf = jl_malloc_stack(&t->ctx.bufsz, t);
-            if (t->ctx.stkbuf == NULL) {
+    if (!t->ctx.started && !t->ctx.copy_stack && t->ctx.stkbuf == NULL) {
+        // need to allocate the stack
+        void *stkbuf = killed && !lastt->ctx.copy_stack && lastt != ptls->root_task && lastt->ctx.bufsz >= t->ctx.bufsz && t->ctx.bufsz <= lastt->ctx.bufsz * 1.5 ? lastt->ctx.stkbuf : NULL;
```

Comment:
Maybe split this condition in a couple lines, this is a bit hard to read

Reply:
I think the size comparison part might not be what's intended? It seems to say `t <= lastt && t <= lastt*1.5` but possibly should be `t <= lastt && lastt <= t*1.5`? (I believe to make sure the reused stack isn't excessively large?)